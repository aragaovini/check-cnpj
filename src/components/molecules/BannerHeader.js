import React from 'react';
import styled from 'styled-components';
import TitleDescription from '../atoms/TitleDescription';
import MaterialIcon from 'material-icons-react';

const BackgroundContainer = styled.div`
    display: flex;
    flex: 1;
    align-items: center;
    padding: 12px;
    background: rgba(165,54,255,1);
    background: -moz-linear-gradient(top, rgba(165,54,255,1) 0%, rgba(116,5,216,1) 90%, rgba(110,0,212,1) 100%);
    background: -webkit-gradient(left top, left bottom, color-stop(0%, rgba(165,54,255,1)), color-stop(90%, rgba(116,5,216,1)), color-stop(100%, rgba(110,0,212,1)));
`;

const AvatarImage = styled.img`
    width: 64px;
`;

const IconContainer = styled.div`
    margin: 18px;
`;

const FlexBlock = () => {
    return <span style={{flex: 1}}/>;
}

const BannerHeader = (props) => {
    const { title, code, image } = props;
    return (
        <BackgroundContainer className="banner-header">
            <IconContainer>
                <MaterialIcon color="#fff" size="40" icon="trending_up" />
            </IconContainer>
            <div>
                <TitleDescription className="title-container" title={title} />
                <div className="code-container">{code}</div>
            </div>
            <FlexBlock />
            <AvatarImage src={image} alt="avatar-profile" />
        </BackgroundContainer>
    );
}

export default BannerHeader;