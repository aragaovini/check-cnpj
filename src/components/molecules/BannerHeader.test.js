import React from 'react';
import { shallow, mount } from 'enzyme';
import BannerHeader from './BannerHeader';
import Avatar from '../../assets/img/avatar.svg';

const BannerHeaderComponent = <BannerHeader title="Nova cotação" code="#0980" image={Avatar}/>;

describe('BannerHeader', () => {

    it('should render without crashing', () => {
        const component = shallow(BannerHeaderComponent);
        expect(component).toMatchSnapshot();
    });

    it('should render the code correctly', () => {
        const wrapper = mount(BannerHeaderComponent).find('.code-container');
        expect(wrapper.text()).toEqual('#0980');
    });

    it('should render the title correctly', () => {
        const wrapper = mount(BannerHeaderComponent).find('.title-container');
        expect(wrapper.text()).toEqual('Nova cotação');
    });

    it('should render one banner', () => {
        const wrapper = shallow(BannerHeaderComponent);
        expect(wrapper.find('.banner-header').length).toBe(1);
    });
});