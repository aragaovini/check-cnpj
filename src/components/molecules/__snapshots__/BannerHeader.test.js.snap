// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerHeader should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BannerHeader
    code="#0980"
    image="avatar.svg"
    title="Nova cotação"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <MaterialIcon
            color="#fff"
            icon="trending_up"
            size="40"
          />
        </styled.div>,
        <div>
          <TitleDescription
            className="title-container"
            title="Nova cotação"
          />
          <div
            className="code-container"
          >
            #0980
          </div>
        </div>,
        <FlexBlock />,
        <styled.img
          alt="avatar-profile"
          src="avatar.svg"
        />,
      ],
      "className": "banner-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <MaterialIcon
            color="#fff"
            icon="trending_up"
            size="40"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#fff",
            "icon": "trending_up",
            "size": "40",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TitleDescription
              className="title-container"
              title="Nova cotação"
            />,
            <div
              className="code-container"
            >
              #0980
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "title-container",
              "title": "Nova cotação",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "#0980",
              "className": "code-container",
            },
            "ref": null,
            "rendered": "#0980",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "avatar-profile",
          "src": "avatar.svg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <MaterialIcon
              color="#fff"
              icon="trending_up"
              size="40"
            />
          </styled.div>,
          <div>
            <TitleDescription
              className="title-container"
              title="Nova cotação"
            />
            <div
              className="code-container"
            >
              #0980
            </div>
          </div>,
          <FlexBlock />,
          <styled.img
            alt="avatar-profile"
            src="avatar.svg"
          />,
        ],
        "className": "banner-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MaterialIcon
              color="#fff"
              icon="trending_up"
              size="40"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#fff",
              "icon": "trending_up",
              "size": "40",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TitleDescription
                className="title-container"
                title="Nova cotação"
              />,
              <div
                className="code-container"
              >
                #0980
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "title-container",
                "title": "Nova cotação",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "#0980",
                "className": "code-container",
              },
              "ref": null,
              "rendered": "#0980",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "avatar-profile",
            "src": "avatar.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
